app/scripts/migrations/096.ts

```diff
@@ -0,0 +1,102 @@
+  meta: { version: number };
+  data: Record<string, any>;
+};
+
+export const version = 96;
+
+/**
+ * This migration will operate the following:
+ *
+ * - Delete `showIncomingTransactions` from `featureFlags` in PreferencesController
+ * - Create a new object under PreferencesController named as `incomingTransactionsPreferences`
+ * 1. which will collect all added networks including localhost
+ * 2. then append the test networks
+ * 3. each of them would become a key coming with the value Ture/False from `showIncomingTransactions`
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+interface NetworkConfiguration {
+  chainId: Record<string, any>;
+}
+
+interface FeatureFlags {
+  showIncomingTransactions: boolean;
+}
+
+interface IncomingTransactionPreferences {
+  [key: string]: boolean;
+}
+
+interface State {
+  PreferencesController?: {
+    featureFlags?: FeatureFlags;
+    incomingTransactionsPreferences?: IncomingTransactionPreferences;
+  };
+  NetworkController?: {
+    networkConfigurations?: Record<string, NetworkConfiguration>;
+  };
+}
+
+function transformState(state: State) {
+  if (
+    !hasProperty(state, 'PreferencesController') ||
+    !isObject(state.PreferencesController) ||
+    !isObject(state.NetworkController) ||
+    !hasProperty(state.PreferencesController, 'featureFlags') ||
+    !hasProperty(state.NetworkController, 'networkConfigurations')
+  ) {
+    return state;
+  }
+  const { PreferencesController, NetworkController } = state;
+  const { featureFlags }: Record<string, any> = PreferencesController;
+  const { showIncomingTransactions }: any = featureFlags;
+  const { networkConfigurations }: Record<string, any> = NetworkController;
+
+  const addedNetwork: Record<string, any>[] =
+    Object.values<NetworkConfiguration>(networkConfigurations).map(
+      (network) => network.chainId,
+    );
+
+  const mainNetworks = [CHAIN_IDS.MAINNET, CHAIN_IDS.LINEA_MAINNET];
+  const testNetworks = [
+    CHAIN_IDS.GOERLI,
+    CHAIN_IDS.SEPOLIA,
+    CHAIN_IDS.LINEA_GOERLI,
+  ];
+  const allSavedNetworks: Record<string, any> = [
+    ...mainNetworks,
+    ...addedNetwork,
+    ...testNetworks,
+  ];
+
+  const incomingTransactionsPreferences = fromPairs(
+    map(allSavedNetworks, (element) => [element, showIncomingTransactions]),
+  );
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
```

Comment:
For future reference, we have this ESLint error because we should be using `@ts-expect-error` over `@ts-ignore`, because `@ts-expect-error` will tell is when the type error goes away.

Ideally we'd use neither of course, but if we really can't fix a type error, that's how we'd silence it.

In this case, it seems that the type error can and should be fixed, so we don't need either.

Reply:
Thanks for suggestion ! i did some object destructuring and solved the warning 🌟